app/routes/questionnaire.py

```diff
@@ -544,14 +546,28 @@ def get_page_title_for_location(schema, current_location, context):
     return safe_content(page_title)
 
 
-def _render_page(block_type, context, current_location, previous_location_url, schema):
+def _render_page(
+    block_type,
+    context,
+    current_location,
+    previous_location_url,
+    schema,
+    questionnaire_store,
+):
     if request_wants_json():
         return jsonify(context)
 
     page_title = get_page_title_for_location(schema, current_location, context)
     session_data = get_session_store().session_data
     session_timeout = get_session_timeout_in_seconds(schema)
 
+    are_hub_required_sections_complete = all(
+        questionnaire_store.progress_store.is_section_complete(section_id)
+        for section_id in schema.get_section_ids_required_for_hub()
+    )
+
+    is_question = block_type == 'Question'
```

Comment:
We may also need to consider `ConfirmationQuestion` block types.

Reply:
We've added "ConfirmationQuestion" now, thanks